src/search/mod.rs

```diff
@@ -95,7 +103,7 @@ pub fn aggregate_query_hits(
             get_smiles_and_extra_data(result, &searcher, smile_field, extra_data_field)?;
 
         final_results.push(QuerySearchHit {
-            extra_data: extra_data.into(),
+            extra_data: serde_json::from_str(extra_data.deref())?,
```

Comment:
`extra_data` is type `String`, right? I think it's more idiomatic to do `&extra_data` and I believe rust will make it in to an `&str`.